packages/epo-widget-lib/src/widgets/CameraFilter/CameraFilter.tsx

```diff
@@ -0,0 +1,123 @@
+import { FunctionComponent, useState } from "react";
+import { SelectListbox, tokens } from "@rubin-epo/epo-react-lib";
+import { useTranslation } from "react-i18next";
+import useResizeObserver from "use-resize-observer";
+import { filters, Filter, rangeConfig } from "./data";
+import * as Styled from "./styles";
+import CondensedFilterRanges from "./CondensedFilterRanges";
+import SpectrumDisplay from "./SpectrumDisplay";
+
+const CameraFilter: FunctionComponent = () => {
+  const { t } = useTranslation();
+  const { ref, width = 1 } = useResizeObserver<HTMLDivElement>();
+  const [activeFilterBand, setActiveFilterBand] = useState<string | null>(null);
+  const { BREAK_TABLET } = tokens;
+  const isCondensed = width < parseInt(BREAK_TABLET);
+
+  const {
+    [isCondensed ? "condensed" : "default"]: { min, max, range: spectrumRange },
+  } = rangeConfig;
+
+  const filterMin = filters[0].range[0];
+  const filterMax = filters[filters.length - 1].range[1];
+  const filterRange = filterMax - filterMin;
+
+  const filtersOnly = filters.filter(({ band }) => band) as Required<Filter>[];
```

Comment:
there is an undefined range in the filters for the gap between the `u` and `g` band which does not have a name but is still included for visual representation. Since our labels are intended only for camera filters, we removed this from the filters before rendering labels.